### ISSUE TITLE
Resize Operator rounds-down instead of round-to-even for int32/uint8

### ISSUE THREAD

**Issue by pranav-prakash** [microsoft/onnxruntime#7368]
**Describe the bug**
As mentioned on https://github.com/microsoft/onnxruntime/pull/7216 I noticed that the resize implementation in ORT _casts_ interpolated values to the destination type instead of rounding. That is, with the current implementation, trying to linearly interpolate to 6 elements between 1 and 2 with an integral destination type will lead to a result of `1 1 1 1 1 2` instead of `1 1 1 2 2 2`. This behavior becomes more pathological (and further from the floating point version) the more values you try to interpolate between two close numbers.

I opened an [issue](https://github.com/onnx/onnx/issues/3390) in the ONNX repo to clarify the intended behavior which is missing from the spec, but given that almost all other onnx operators use round-to-even mode I suspect that the current implementation in ORT is a bug.

Changing the current implementation for `uint8/int32` would be a breaking change however.

**Urgency**
P2 urgency; we need this behavior clarified so that we can run a quantized mask-rcnn model.

**Comment by hariharans29** [comment]
yeah unfortunately it will be a breaking change and I think it is better to have the onnx spec clarified and have the op versioned and fix the implementation for its next version.
